Overhaul themes and intro content

WasabiFan opened the pull request:
## Current TODO list

- [x] Delete unused CSS/JS files and HTML links
- [x] Figure out logo
  - Want to use original shapes, just change color
  - Decide on a header design which makes the logo big enough to see it
  - Auto-shrink logo into small version?
- [x] Add favicon of logo
- [x] Add more images to project featurette image
  - Contact authors of images we want to use to make sure it's OK
- [x] Downsample final projects pic to make it less insanely large
- [x] Change text on home page to make it less like a lie (it'll probably take more than a minute)
- [x] Fix quick nav so that it doesn't actually submit to server when enter is clicked
- [x] Adapt custom alert styles for panels
- [x] Wrap YouTube embed in Bootstrap embed styles

## Original post

To start off, I'm going to make it clear that this is a prototype, and the current state of this code is not something that should ever see the light of a production codebase. I'm just trying to get it out there for some early feedback.

So, this is a set of changes that I have wanted to make for a while. It replaces the hacks that I have introduced over the past 20-something PRs as well as the styles that we inherited from one of GitHub's getting-started templates with Bootstrap, a CSS framework that gives you a fully-responsive website and a (I dare say) beautiful default template to start with. It makes the site fully responsive, so the content flows as you resize the window. Along with this, it brings in a 12-column system and lots of CSS classes to help organize pages of content. In doing this, I started by stripping out all of the other CSS and starting with a test CSS file to experiment. That means that, until I get to fixing a page, it will look be horribly broken.

So far, I have added bootstrap and re-done the index page so that I (we) can judge this and see if we like it. I wouldn't recommend visiting any other pages yet. I took a fair amount of the existing content from the intro page and added some to fit the new form-factor.

**I have made these changes live [here](http://wasabifan.github.io/ev3dev.github.io/)** so that you can check it out. My main question is this: Should I put in the extra work of fixing the rest of the site, or do we want to stick with the site that we already have? Personally, I think that this is a great improvement that makes ev3dev look much more like a usable project, and if I follow this through there will be a clear path for a new user to learn, download, and contribute. If we decide to keep the existing design and code, I'll open a new PR to just update some of the typos on the home page and then add the better download path and repo information.

Currently, these are the things you should look at: 
- The home page
- The project index page (list of projects) 

And this is what I haven't fixed yet (so shield your eyes):
- Pretty much all other pages
- Project detail pages
- Some styling and sizing on the nav header
- The footer
- _quick nav_

ddemidov commented:
Just an opinion: it does look more professional, but at the same time a bit 'sterile'. May be I am just missing the old background color. Also, the top menu is not visible in the mobile browser (chrome for android).

WasabiFan commented:
> it does look more professional, but at the same time a bit 'sterile'. May be I am just missing the old background color.

Yeah, same feeling for me. I think that part of that is the navigation bar, and part is the drastically different color scheme. All the colors are easily theme-able, so I am planning to look into available dark themes to see how they look.

> Also, the top menu is not visible in the mobile browser (chrome for android).

I was noticing mobile nav issues too... I think I didn't import the JavaScript properly that is supposed to manage the transition. I'll take a look at that and try to fix it -- thanks for the test case.

WasabiFan commented:
I think I can also make it 'feel' better by adding color, like the green button, throughout the page. I'm considering a carousel control that has a rotating set of pictures from projects, which would definitely 'spruce up' the look of the index page. That will be a bit more intense than most other styling work, so I think I'll tackle that once I get the go-ahead and the rest of the pages are fixed.

rhempel commented:
- This is looking pretty cool! I think that a "darker" color scheme would look a bit more techno but that's just CSS styling so it's no big deal to change - choosing the colors is different :-)
- I'd like to see the ev3dev.org logo a bit bigger in the header, and maybe key some of the color scheme for the site from the logo.
- If possible, look at a font like ORBITRON for the "EV3" text to give it a more robotic feel.
- Add a sectoin on the front page that explains that many popular scripting languages are supported right out of the box.
- Add a section on the front page that indicates ev3dev is a painless way to learn more about Linux?

rhempel commented:
Oh, and if you don't already have one, I'd be happy to send you some custom ev3dev.org tiles!

dlech commented:
I guess I kind of like sites that don't use bootstrap just because they look different. There are so many sites these days using bootstrap and they all look basically the same. But that can be a good thing too, I suppose.

The new front page is certainly much more inviting.

I'm not a fan of the carousel. I don't like web pages that move (unless I tell them to).

WasabiFan commented:
> I guess I kind of like sites that don't use bootstrap just because they look different. There are so many sites these days using bootstrap and they all look basically the same.

I think the issue there is the allure of using default bootstrap themes and not customizing anything: people just kept taking advantage of the ease-of-use. My hope is to take one of the basic themes as a starting point and then customize it to look more "techno". @rhempel and @dlech what do we want here? Dark theme, with a techno font? And do we want the fixed navbar at the top, or the page-scrollable bar that is detached from the top and sides?

> The new front page is certainly much more inviting.

That's what I was going for! My hope was to make it easier for someone to get started from looking at the website, and I think the home page is the first step toward that.

> I'm not a fan of the carousel. I don't like web pages that move (unless I tell them to).

I was imagining more of a "static carousel" thing... OK, maybe that doesn't make any sense ;). I mean some sort of a showcase of images and an explanation of projects. I'll try it out and report back with a prototype.

So, what I'm hearing is this: It looks better, but the theme needs some work. I take that to mean that I should continue to see how it plays out -- is this correct?

My first task will be to get that styling figured out and enlarge the nav, and then I'll add that extra content that @rhempel suggested.

Thanks for the feedback!

WasabiFan commented:
@dlech I just updated the PR with that projects part that I mentioned along with the updated content from @rhempel's suggestions. Working on theme now.

WasabiFan commented:
I just pushed some updates that make some pretty big changes.

Primarily, it switches to a dark theme, which it sounded like you'd like much more. Along with this I changed the nav bar to be a flavor of "EV3 orange" that works well to cover a fair amount of the screen without giving you permanent eye damage.

The other big change that I made is more of an experiment than anything else. I realized that the normal ev3dev logo is pretty hard to read when put in the bar (or as a favicon). I made a test icon that is just a cropped version of the normal logo -- it's the "D" in "dev". I think that, at that size, it looks better than the whole logo (and same at the favicon size), but that "D" that I came up with still isn't very good. @rhempel do you have any ideas for ways of showing the logo at smaller sizes?

So, how is this? Does it seem like it has deviated from the standard Bootstrap look enough, while still maintaining the professional appeal? And, should I investigate the full-on "techno" fonts?

Notes on to-do items:
- The hover colors on the nav are very funky right now. If we decide to stick with the orange, I'll have to change that.
- I haven't yet fixed any other pages

ddemidov commented:
This looks much better to my taste. I would change the link color (green looks not that great on both orange and gray backgrounds). Regarding the logo, the 'D' does not look recognizable. How about filling its background with gray for the nav bar, so that orange letters would still be readable? Something like this:
![logo](https://cloud.githubusercontent.com/assets/270503/8518317/81d5942e-23d1-11e5-811d-3dac401b40a9.png)
Or may be with border and background colors switched.

WasabiFan commented:
I was thinking about that, but couldn't read the text at that size, which seemed to void the point of the logo. Either way, I agree, the "D" isn't a good option -- I just decided to commit and push what I had instead of working more on getting the real logo to work at that point.

I recommend that you open the dev tools in your browser and replace the 'src' of the logo image with the GitHub-based URL that was there before so that you can see what it looks like and experiment with the version that you showed. With that gray background, I think it might actually be readable... But I'm not sure.

And yes, the green links are completely hideous :wink: I think I'll make them a shade of orange.

ddemidov commented:
> I recommend that you open the dev tools in your browser

Yes, I can see that now. It does look like the logo size could be increased without changing the navbar height (I have no idea how though):
![sshot](https://cloud.githubusercontent.com/assets/270503/8518566/ee73e868-23d3-11e5-8a29-ee302bf5ed09.png)

cavenel commented:
In stylesheets/bootswatch-darkly.css, you can change line 4328 from `padding: 19.5px 15px;` to `padding: 5px 15px;`. Then it looks nice even with the original logo.

![ev3devlogo](https://cloud.githubusercontent.com/assets/6164758/8519215/67386b50-23d1-11e5-9cae-1f6380cab066.png)

Else I guess it is on the todolist for later but the project pages are really buggy ;-) And it shows even more why we should not have fixed height for Youtube videos!

WasabiFan commented:
@cavenel that looks like the right place to change it -- and it does seem to make it much better. I think it still makes it look a bit weird at that size relative to the rest of the navbar, but I'll play around with it and see what I can do. I think that, based on the image you provided, it shouldn't take too much to get the logo to be easily readable.

@rhempel would you be willing to give me the source files for the logo so that I can try out some different background colors? We have been using the version from the organization page and it's pretty low-resolution even for tests.

WasabiFan commented:
@cavenel I have been visiting the project pages every few hours as I get frustrated with something and need a laugh :wink: They're the most messed up pages now that I removed their CSS.

dlech commented:
@WasabiFan, I have pushed the official logo SVGs to the meta repo: https://github.com/ev3dev/ev3dev/tree/master/logo

WasabiFan commented:
I just pushed an update that takes another attempt at the logo. I spent a while trying to get the unmodified logo to work (thanks @dlech for the files!) but it's just too small when put in any reasonably-sized nav bar. So I tried to go for a "mostly unmodified" version that's just the core "ev3dev" text from the logo. This is _much_ more recognizable than my earlier attempts, and it is very easy to read.

So, how's this version?

rhempel commented:
Thanks @dlech for making the svg files I sent you available to @WasabiFan - just the core ev3dev is ok, but I'm not sure I understand the issue with pushing all-white ev3dev.org in the navbar.
I do like the look of the new pages - let's put a pin in the logo placement for now and get the rest of the site updated - the projects main page looks pretty nice!
The special characters leading in the headings on the main page are not rendering properly for me in IceWeasel (Firefox).

WasabiFan commented:
> I do like the look of the new pages - let's put a pin in the logo placement for now and get the rest of the site updated - the projects main page looks pretty nice!

The project detail pages should be fixed now as well. I still need to make the YouTube embeds responsive, but everything else should be pretty much done. 

> The special characters leading in the headings on the main page are not rendering properly for me in IceWeasel (Firefox).

I'm trying this in the most recent version of FF and I can see the icons on the home page. I am, however, seeing multiple errors like this in the console:
```
downloadable font: download failed (font-family: "Glyphicons Halflings" style:normal weight:normal stretch:normal src index:1): status=2147746065 source: http://localhost:4000/fonts/glyphicons-halflings-regular.woff2
```

And the logo area in the navbar is wider than it is in any other browser. How does it look for you? Do they not show up altogether, or is it misaligned? And do you see those errors in the console?

ddemidov commented:
> The project detail pages should be fixed now as well.

The project pages still look a bit weird in chromium  (Arch linux):
![ppage](https://cloud.githubusercontent.com/assets/270503/8567631/a82338a8-256f-11e5-9998-b4c2b9445f44.png)

WasabiFan commented:
I think the weirdness there is just the extremely large icons... I'll need to figure out why they are so big (and, how I missed it). Anyway, it should be a simple fix... Unless it is a browser compatibility issue :(

ddemidov commented:
> I think the weirdness there is just the extremely large icons.

That, and also the aspect ratio of the youtube preview (or is it just a widescreen version?).

ddemidov commented:
Ok, I see that the actual playback goes with the correct aspect ratio, so its just icons.

WasabiFan commented:
> That, and also the aspect ratio of the youtube preview (or is it just a widescreen version?).

I do plan to change this; Bootstrap has a method of responsively embedding YouTube videos, so once that's in place it should maintain the proper aspect ratio. Thanks for reminding me about this -- had completely forgotten to fix the embeds!

WasabiFan commented:
Here's my checklist of the big things that still need to be done to make this fully usable:
- [ ] Add downloads page
- [ ] Add GitHub info page
- [ ] Fix docs
  - [ ] Fix docs index page
  - [ ] Fix other "list" pages (like "drivers")
- [ ] Update "share" and "contribute" pages
- [ ] Fix quick nav styles
- [ ] Figure out formatting for TOCs
- [ ] Center buttons on index page

I'll then have to go through all other pages and make sure that they look okay with the new styles.

@ddemidov I think I have fixed the project pages now. The YouTube embeds aren't responsive yet, but everything else should be better (the icons and footer).

Is there anything else that anyone has seen that I need to fix? I'm sure there are more broken pages, but these are all the big ones that I remember.

ddemidov commented:
Yes, I think the project pages look fine now. Here are some small quirks I noticed:

*  Tables look a bit crowded. Compare new supported motors [page](http://wasabifan.github.io/ev3dev.github.io/docs/motors/) with the old [one](http://www.ev3dev.org/docs/motors/).
* Is it possible to change color scheme for the code blocks? I think they are too bright on the dark background: [new](http://wasabifan.github.io/ev3dev.github.io/docs/getting-started/) vs [old](http://wasabifan.github.io/ev3dev.github.io/docs/getting-started/).

WasabiFan commented:
> Here are some small quirks I noticed

You're completely right. I have added those two items to my list.

dlech commented:
@WasabiFan, are you modifying the .less files or are you modifying the .css directly?

WasabiFan commented:
> @WasabiFan, are you modifying the .less files or are you modifying the .css directly?

I'm modifying the .less files from bootswatch and recompiling. After I get the styles closer to finalized, I'll get the Less customizations integrated in a way that they're easy to build.

dlech commented:
Awesome.

WasabiFan commented:
I have pushed some commits today that continue to clean up some pages and fix broken elements. Now the quick nav doesn't expand the whole header, code blocks don't blind you and we don't load two copies of Bootstrap on every page. I also updated the getting started page to take advantage of Bootstrap's alerts and panels.

My next step will probably be to update the barebones docs intro page, and then add the new downloads and GitHub info pages.

WasabiFan commented:
@dlech It looks like Bootstrap provides a good table style, but it only applies to elements with the `.table` class... so we need to modify the actual HTML to add the required classes. Most of the tables on the site are in docs pages that are generated by your Python scripts, so I can't just modify them in the HTML in this repo.

What would you like to do about this? Making the changes in the other repos that house the docs will mean that the classes aren't just added to my WIP branch, but to the main one as well if you re-generate it.

If you're interested, the classes that I want to add to our tables are `table table-striped table-bordered`.

dlech commented:
The tables on the individual sensor and ports pages are generated by a liquid template from json data, so you should be able to just update the templates.

* [Sensor](https://github.com/ev3dev/ev3dev.github.io/blob/master/_includes/sensor.md)
* [Port](https://github.com/ev3dev/ev3dev.github.io/blob/master/_includes/port.md)

I don't think there are actually any tables generated directly in the python code, but if there are, we'll just figure it out after this is merged and I do a kernel release. For the time being, just update every instance of table here whether it is automatically generated or not.

dlech commented:
And is it just me, or does the orange navigation bar just not go with the rest of the colors?

WasabiFan commented:
> For the time being, just update every instance of table here whether it is automatically generated or not.

OK; sounds good.

> And is it just me, or does the orange navigation bar just not go with the rest of the colors?

I'm still trying to figure out the color scheme. I think that the orange is OK, but it starts looking funky on some monitors. And the blue "info" colors get very ugly with the orange links and nav -- I still have to change that.

Do you think that we should switch it to a different color, or tune the orange theme?

dlech commented:
Have you tried making the links white and underlined as in the booswatch themes? Also, the "warning" button color could be made more yellow. Although we don't really have a need for it anywhere that I know of. There are several places where it is being used where I think a primary or default button theme should be used instead.

WasabiFan commented:
I'm not sure why I used the warning class originally, but it should be fixed now.

> Have you tried making the links white and underlined as in the booswatch themes?

I'm not quite sure of what you mean. Looking at the Bootswatch example pages, most of the styles use the theme color of the style for the links. I can get a sample of what white links look like on our site, but there isn't any shade of white/black/gray that I could find that worked well with the surrounding colors. White links do look good in alerts, though.

dlech commented:
Apparently I don't know what I'm talking about. You are right about the link styles. I guess I was just looking at the links in the alerts and not in "normal" usage. So what if we just make the links white in the alerts and leave them as-is elsewhere?

WasabiFan commented:
> So what if we just make the links white in the alerts and leave them as-is elsewhere?

Sounds good; I'll make them white.

WasabiFan commented:
@dlech Turns out that, like a lot of other things in Bootstrap, there's a class for alert links that I wasn't applying. It makes them white, as you had thought, and now they look pretty good.

I also went through and added icons to all the alerts, fixed the mobile menu, and updated the screenshots to float inline with a caption. In doing this I added a few new items to my to-do list above, but most of them are small -- so I should have it done fairly soon.

As for the new GitHub and downloads pages, I have decided that I want to get this fairly large PR done before I start adding completely new content. But I have figured out the proper way to get Jekyll to give us `site.github` on local development servers, so I know that it's feasible to work with.

P.S. When I enable the gem that gives us that GitHub metadata when served locally, all the links suddenly point to the real site instead of the local version because we use `site.github.url` to fix relative links. I think I'll have to change all instances of `site.github.url` to our own "site root" variable which we disable locally. Still trying to figure that out, though.

dlech commented:
Does `{{ site.baseurl }}` work instead of `{{ site.github.url }}`?

WasabiFan commented:
@dlech are the tabs generated by the kramdown engine? If not, is it something that we are doing ourselves? I see things like this:
```
{: tab="Mac OSX"}
```

to denote tabs, but don't see anything in our code or in the kramdown docs that explain how this works.

WasabiFan commented:
> Does  {{ site.baseurl }}  work instead of  {{ site.github.url }} ?

It looks like `site.baseurl` is just a recommended name for a variable to use to prefix your paths so that you can change it on-the-fly. It doesn't look like GitHub sets this itself, so any configuration of that path must be done manually. If we set it to `/ev3dev.github.io`, we would break the main site, and if we set it to a blank string we'd break the user test instances.

So, I think that your original solution of using `site.github.url` is the closest one: it is always correct as long as it is being hosted on GitHub. But, once we host it locally, the available metadata integration services make it use their paths and make it very difficult to navigate between pages for testing. I'll look into trying to override their properties, maybe using a combination of `baseurl` and some extra frontmatter settings.

dlech commented:
Re: tabs, this is the [attribute list definitions](http://kramdown.gettalong.org/syntax.html#attribute-list-definitions) feature of kramdown. You can use it to apply html attributes to markdown. I noticed there were several markdown pages that you converted to html, presumably because you needed to apply classes for bootstrap. You can achieve the same thing this way. In cases where there is lots of content, I would prefer markdown to html.

WasabiFan commented:
> I noticed there were several markdown pages that you converted to html, presumably because you needed to apply classes for bootstrap. You can achieve the same thing this way. In cases where there is lots of content, I would prefer markdown to html.

In most of these cases, I was switching it because I had to wrap a considerable portion of the document in a div or other element, which caused it to ignore any markdown syntax inside the element. I'll check the history and make sure that there were no unneeded conversions where I needed to just add a class.

dlech commented:
There is a way around this too :wink:

Read the end of the [html blocks](http://kramdown.gettalong.org/syntax.html#html-blocks) section.

WasabiFan commented:
Wow... I didn't realize that kramdown had so many extra features! I'll go through and undo the conversions (after reading their docs, of course :wink:)

WasabiFan commented:
@dlech That kramdown docs page says that adding `markdown="1"` to an HTML element will make it parse it as Markdown. This seems to work, but it takes the indentation of the text inside the HTML as code formatting and puts the whole thing in a code block. Do you know of any way to have it ignore that indentation?

dlech commented:
The indentation is part of the markdown syntax, so there is not anything you can do about it. I would suggest just not indenting html on markdown pages.

WasabiFan commented:
@dlech I just went through and updated the tab scripts and associated declarations to use our page theme and Bootstrap's layout. One of the pages (the main one that I was testing with, [Using USB Tethering](http://wasabifan.github.io/ev3dev.github.io/docs/tutorials/using-usb-tethering/)) seems to work pretty well. But on all the others (like [this one](http://wasabifan.github.io/ev3dev.github.io/docs/tutorials/connecting-to-the-internet-via-usb/)), where there is lots of content, the whole thing falls apart.

Large chunks of instructions are rendered in code blocks, there are unmatched HTML tags floating on the page, and the tabs don't actually show up. When I disable the tab script, I can see that the issues come from the markdown-to-html conversion, and not the tab conversion on the client-side. All that I changed was the names of the classes and attributes that were used in the HTML, and I don't see any other modifications in the diff. Yet it's obvious that it was something that I changed... do you have any ideas?

dlech commented:
Sorry, I won't be able to look at this for a week.

rhempel commented:
I'll try to get at this towards the end of the week - just finishing up the last few days of vacation...

WasabiFan commented:
I think I have figured out at least part of the problem with the tabs... the screenshot include is HTML and doesn't play well with the markdown page that it is injected into. Based on the behavior that I am seeing, the HTML is injected into the markdown page _before_ the markdown is rendered, and some of the elements that are used aren't parsed correctly by kramdown.

@dlech any ideas on how I can deal with this? Should I make a separate screenshot include for markdown?

dlech commented:
Add `markdown="0"` to the `<span>`? Or, add a newline in the include file before the first line if it is an indentation problem?

WasabiFan commented:
The problem is that the first line of the HTML is indented from the indentation of the include in the markdown files, while the rest of the HTML isn't indented at all. It works if I manually indent all but the first line, but that only works for some pages and is pretty hacky.

The core of the problem is that we are trying to use HTML in a markdown list. Looking through the tutorial pages' content, I don't see much that benefits from the markdown syntax; can I just convert the two or three pages that have screenshots in lists to HTML? It will make modifying the text easier (because you don't have to manage indentation) and make it easier to read overall, due to the nested lists.

WasabiFan commented:
I think I want to remove my tab changes from this PR (use the old ones) for now so that I can just clean it up, take final feedback, and figure out what I need to do to get this merged. Aside from the tabs on the tutorial pages, my changes accomplish everything that I had hoped to; The site looks better, the intro content is organized cohesively, and it's easier to find what you're looking for.

There are a few lasting things that I think we should figure out:
- This currently uses the test version of the logo in the header. What do we want to do here? Should I change it to the unmodified version, or do we want to use this? If we want to use the modified version, we should take the source files and add them as a "simplified" version of the logo for colored backgrounds in the main repo where the other files are.
- The "projects" background image from the homepage is currently a PSD (Photoshop) file that I haven't checked in to the repo. Should I check it in as-is, or convert it to a file type for an open-source Photoshop equivalent?
  - The file will likely be fairly large. Is it time to investigate GitHub's [LFS extension](https://git-lfs.github.com/)?
- Whatever the decision is on the first point, we should convert _a_ logo to `.ico` format to act as a favicon.

Aside from those housekeeping items, I really just need a review of the whole thing -- there are probably some typos, dead links, etc. The most recent changes have been pushed to my fork's GH pages site [here](http://wasabifan.github.io/ev3dev.github.io/).

dlech commented:
About the logo:

* I would like to keep the `.org` part on the logo. I think it's OK to change the color, but 
for consistency, we should keep everything else the same.
* I know part of the reason for changing it was for readability - using bootstrap magic, could we make the logo start larger and get smaller when you scroll down?

About the "project" background:

* I don't think we need to include the photoshop source file in git.
* Could we add some more photos so there are not repeated images (well, at least not as many)?
* Have you asked permission to use these photos?

About the home page:

* The menus have the same background color as the first section, which makes them hard to see.
* In light of #396, maybe instead of saying "See what *you* can do with ev3dev", say "See what others have done with ev3dev".
* "It'll only take a minute" sounds good, but is unfortunately far from the truth. We should think of something else to say there.

I'll send a pull request with some changes I would like to make on the home page and I'll add more comments here if I find anything else as I go through the rest of the pages.

dlech commented:
Info bars: The blue looks strange to me with the rest of the color scheme. What about using `#111111`?

dlech commented:
I guess technically, those are "alerts" not "bars". I'm looking at the other ones not too and they all seem very bright. Can we make them all a bit darker? (Blue might be fine this way.) Also, can we make the "Danger" a bit more red instead of pink or salmon or whatever you want to call it.

rhempel commented:
I agree with @dlech - please keep the .org in the name

dlech commented:
I like how have rearrange the docs a bit and made a submenu for it.

The `docs` link in the breadcrumbs aren’t' working, but it looks like an issue with it not including the correct base url and will probably work once we merge.

I want to make the [sensors](http://wasabifan.github.io/ev3dev.github.io/docs/sensors/) and [motors](http://wasabifan.github.io/ev3dev.github.io/docs/motors/) pages really easy to find, so they should be on the submenu too.

rhempel commented:
Maybe it's the tone of the blue - the blue orange combination is a standard contrasting colour scheme, as evidenced here [http://www.slashfilm.com/orangeblue-contrast-in-movie-posters/]
That being said, is it possible to add an orange outline (instead of black) on the dropdown menu - it might make the dropdown easier to see.
Overall the changes make the project look much more modern, we really appreciate the work you're putting into this - got your tiles yet?

dlech commented:
Maybe I better start putting images in news articles. The news page looks pretty boring. It's OK for now, but just wanted to throw that out there in case anyone else has any good ideas.

dlech commented:
Share page: can we put the different sections in boxes with borders or something like that to help divide the content. @rhempel, do we want to add a link to our facebook page here or are we not ready to advertise it yet?

dlech commented:
Help page: This page looks a bit crowded around, particularly around the "Search Issues". Since these are basically, 1,2,3... steps here, maybe we could use an ordered list to divide things up.

rhempel commented:
I put the Facebook page there as a placeholder before someone else nabbed it. I'd be happy to give anyone that has ev3dev.org News page privileges the ability to post to the Facebook page :-)

WasabiFan commented:
> I would like to keep the  .org  part on the logo. I think it's OK to change the color, but for consistency, we should keep everything else the same.

I'll open up the original again and see how it is after making it white.

> I know part of the reason for changing it was for readability - using bootstrap magic, could we make the logo start larger and get smaller when you scroll down?

That's a cool idea; I don't think that there's a pre-made bootstrap formula for that, but it shouldn't be too hard with JS (and there might even be a CSS way of doing it). I'll just need to make sure that I disable it in mobile view.

> Could we add some more photos so there are not repeated images (well, at least not as many)?

We can; I chose the most interesting (both intellectually and visually) for a new user to include in that image originally, but I believe that there are two or three more that I can add. I did my best to distribute them throughout the image, but obviously it is still pretty noticeable :wink:

> Have you asked permission to use these photos?

Most of the projects on the ev3dev projects page have either an MIT license or the GPL in their source repo. There are still a few that don't have a license listed, so (based on my limited "open-source law" knowledge) those are technically completely restricted by assumption.  Should I post to the respective repos and confirm permission (or mention them here and see if they respond)?

> The menus have the same background color as the first section, which makes them hard to see.

Hmmm... I can try some other colors and see what stands out. I like @rhempel's suggestion of adding an orange border, so I'll try that as well.

> ...instead of saying "See what you can do with ev3dev", say "See what others have done with ev3dev".

Will-do.

> "It'll only take a minute" sounds good, but is unfortunately far from the truth. We should think of something else to say there.

Yeah... that is a bit of a lie :wink:. I can't come up with any good subtitles for that one... will have to let it germinate for a bit.

> I want to make the sensors and motors pages really easy to find, so they should be on the submenu too.

I'll add those links to the docs menu.

> .. I'm looking at the other ones not too and they all seem very bright.

Yeah, the alerts are definitely over-powering (especially when there are multiple visible at once. I like the colors themselves, though; I'll see how they are if I darken them, and I'll also try a border of the bright color with gray in the middle.

> Also, can we make the "Danger" a bit more red instead of pink or salmon or whatever you want to call it.

Sure; I'll make the danger look more danger-y :wink:

> Share page: can we put the different sections in boxes with borders or something like that to help divide the content.

I'll try adding some `<hr/>`s (and equivalents) in that area to see if I can break it up.

> Help page: This page looks a bit crowded around, particularly around the "Search Issues". Since these are basically, 1,2,3... steps here, maybe we could use an ordered list to divide things up.

I've found that lists with that much content in them often make the page look pretty awkward. I'll add some spacing and maybe some titles, and compare it to a version using a list as well.

### TODO List
- [ ] Try a white version of the original logo
- [ ] Try an auto-shrink logo
- [ ] Add more images to project featurette
- [ ] Make drop-down menus more visible
- [ ] Change project featurette text
- [ ] Fix the "getting started" lies
- [ ] Add links to sensor and motor docs pages
- [ ] Make alert colors less seizure-inducing
- [ ] Make danger colors look less pink
- [ ] Add clear structure/sections to "share" page
- [ ] Un-crowd the "help" page

rhempel commented:
If you need help with getting the svg file for the logo to work better let me know. I prefer to have all of our graphics done in svg and rendered/exported to the desired size/format from InkScape (or whatever svg tool we like) I'd prefer NOT to render a standard logo to png/jpg and then start monkeying around with pixels.

G33kDude commented:
Why even bother with exporting? All major browsers render svg fairly well.

dlech commented:
> Most of the projects on the ev3dev projects page have either an MIT license or the GPL in their source repo. There are still a few that don't have a license listed, so (based on my limited "open-source law" knowledge) those are technically completely restricted by assumption. Should I post to the respective repos and confirm permission (or mention them here and see if they respond)?

The photos are most likely not part of the source code. You should be able to find email address for the respective owners from the git commits to ev3dev.github.io. Or, if you want to use GitHub mentions, I would suggest opening a new issue.

rhempel commented:
I was just assuming that the issue was not .svg :-) And you know what assuming does.

WasabiFan commented:
>  I prefer to have all of our graphics done in svg and rendered/exported to the desired size/format from InkScape (or whatever svg tool we like) I'd prefer NOT to render a standard logo to png/jpg and then start monkeying around with pixels. 

I agree -- I have been making sure to keep source copies of all the images that I include, both for the logo (SVG) and the project images.

> Why even bother with exporting? All major browsers render svg fairly well.

SVG behave similarly to raster images when included with an image tag, but I've come across cases where they ignore scale or do other weird stuff like that (mainly because widespread support is relatively new). I'll try including it as an SVG and test to make sure that nothing explodes.

> You should be able to find email address for the respective owners from the git commits to ev3dev.github.io. Or, if you want to use GitHub mentions, I would suggest opening a new issue.

I'll send them an email to confirm.

ddemidov commented:
>> "It'll only take a minute" sounds good, but is unfortunately far from the truth. We should think of something else to say there.

> Yeah... that is a bit of a lie :wink:. I can't come up with any good subtitles for that one... will have to let it germinate for a bit.

Why not put a bank-style small-text footnote here with a reassuring joke? Something like `It will actually take more than a minute, but due to high fun factor, you most probably won't notice it`?

WasabiFan commented:
I have pushed some commits to attempt to address some of the points in the list above:
- Added the sensor/motor docs links to the drop-down
- Spaced out the 'support' page a bit
- Made the 'danger' color more 'danger'-y
- Made the background alert color darker, and added some extra styling to make it look better
  - Haven't updated panels yet

So, how is it?

WasabiFan commented:
@dlech I just pushed your merged changes along with some minor edits to my `gh-pages` branch.

dlech commented:
Looks good. You missed a `<sup>Ⓡ</sup>`.

G33kDude commented:
Isn't `<sup>&reg;</sub>` usually preferred for greatest compatibility?

dlech commented:
Thanks for the tip. That sounds like a good idea.

WasabiFan commented:
Updated.

dlech commented:
@WasabiFan, I know you have been busy with other things lately, but I would like to get this merged before our next image release. Could you at least copy your todo lists from [here](https://github.com/ev3dev/ev3dev.github.io/pull/70#issuecomment-120517291) and [here](https://github.com/ev3dev/ev3dev.github.io/pull/70#issuecomment-143894391) to the first comment and make sure they are up-to-date so I can see where we are and what is left to do?

WasabiFan commented:
Sure; It's currently 11:00 in Seattle (where I live) so I need to get to sleep, but tomorrow morning I'll compile and update the lists and probably make some progress on the tasks that are still outstanding. Thanks for being so patient as I keep half-finishing things and then forgetting about them :wink:

Overall, I think the majority of the site updates are complete; I think it's just the logo, home page, and project image at this point, as well as a final review.

WasabiFan commented:
@dlech I added the current TODO list to the first comment in this PR to make it easily-accessible. I can get the alert styles added onto panels pretty fast, so I think I'll do that now.

WasabiFan commented:
How about instead of saying "It'll only take a minute", we say "It'll be worth it"? Or, if we don't think it's improving the message, we can remove it altogether.

dlech commented:
Yeah, I still haven't thought of anything good to say there. For now, how about we just take it out and make the text on the "Projects" and "Get Started" buttons bigger.

dlech commented:
Or you could say "Grab your bot and let's go!"

WasabiFan commented:
I just removed that subtitle and enlarged the button text; how's that?

dlech commented:
The buttons look better. That section just feels like it needs something else though.

WasabiFan commented:
@dlech Here are the options for that text that I have so far (some of them are from our discussion above, others of varying quality as I go):
- "Grab your bot and let's go!" / "Grab your EV3 and let's go!"
- "It'll be worth it!" / "You'll be glad you did"
- "Check out the getting started guide"
- "We'd love to have you"

Do any of those sound good to you? I was considering trying an alternative to the muted subtitle (some other visual element), but nothing that I found looked very good.

dlech commented:
I feel a little embarrassed picking picking my own, but out of the list, "Grab your bot and let's go!" is the one that jumps out at me.

WasabiFan commented:
That's fine :wink: I was just trying to add more to make sure we had evaluated a few options. Honestly, that would've been my choice as well.

WasabiFan commented:
BTW, I'm currently working on the project banner image -- should have that committed soon.

WasabiFan commented:
@dlech I just sent an email to the contributors of the project images asking for permission to use them. I also CCd you so you could see it.

@ensonic The majority of your commits use an @google.com email address, which is a corporate Google email (not Gmail). I wasn't sure if that was a mistake (supposed to be @gmail.com) or if you are actually a Google employee, so I decided not to include you on the 'to' line initially -- do you have a public email address that you'd like me to use?

G33kDude commented:
As much as I like group emails, I think that's what the BCC field is for. Anyways, shouldn't GitHub ping me if you just mention my @name? Also, you can find me anytime in the IRC channel.

WasabiFan commented:
> As much as I like group emails, I think that's what the BCC field is for.

I thought about that, but given that it's all public info anyways, decided that there was no reason to hide it.

> Anyways, shouldn't GitHub ping me if you just mention my name?

Yes, but (as per the conversation with @dlech above) we decided that email would be more well-suited to the task.

The idea was to get the confirmations in one place to have it to reference later if needed (although I doubt that it'll be needed, it's best to be safe).

> That includes you, I suppose?

It's not as much me that you're permitting as it is ev3dev as a whole. We're just trying to cover our metaphorical bases.

rhempel commented:
@wasabifan - can we consider adding a notice that proiject submitters must click yes or no to allow the ev3dev project to use their robot images (with proper attribution of course). That gets around having to ask for permission later. Not too big a deal now, but we'll be getting more submissions as time goes on.
This is what the LDraw community ended up doing - by submitting your project you agree to let us use the pictures.

WasabiFan commented:
I'd say that's the way to go; I think the simplest solution there is to ask authors of posts to confirm permission in their PR comments before an admin merges it. We don't even need a notice with buttons per se as long as we ask them -- there's already the administrative process of accepting a PR that one must go through.

That does, of course, require the maintainer (most likely @dlech) to ask them to confirm before merging, which could introduce some human error :wink:

WasabiFan commented:
@dlech I just pushed some changes which add a white version of the standard ev3dev logo to the nav bar. How is it?

ensonic commented:
@WasabiFan Yes I work for Google and the company prefers we using my corp email address to emphasize Googles support of open source projects.

WasabiFan commented:
@dlech Last design decision to nail down before I'd say this is ready for a final review: What should the favicon be? The whole logo looks like a pixelated mess at that miniscule scale, so I think it'd need to be pretty simple -- a single shape or letter. Any ideas?

dlech commented:
I was thinking just the `E` from the logo.

WasabiFan commented:
I made a test of that -- here's what it looks like:
![snip_20151116201602](https://cloud.githubusercontent.com/assets/3310349/11202631/f33d14c6-8c9e-11e5-9019-2e89db5678fb.png)

...and with a margin:
![snip_20151116202622_margin](https://cloud.githubusercontent.com/assets/3310349/11202744/58331f6e-8ca0-11e5-9d2e-f0bd1a64cab2.png)

I think that one with the margin is pretty good. Should I go with that?

rhempel commented:
I like the one with the margin - nice job on all of this - looks great!

WasabiFan commented:
As of now, I declare this feature-complete -- barring a review, I believe that this is ready to be merged.

rhempel commented:
Wait! What about an animated favicon where ev3dev.org scrolls :-)
[http://www.animatedfavicon.com/]

WasabiFan commented:
Although the concept is interesting, I think we could ~~piss off~~ frustrate users with an animated favicon :wink:

In general, animated things -- especially at that scale -- are pretty hard to read, anyway. And as of now browsers are starting to remove support for animated icons from what I have read.

rhempel commented:
I get it :-) It's kind of like the goofy <marquee> tag that HTML used to have - never a big fan of it.

Let's go with what you have done so far - I really like the dark look.

dlech commented:
Somebody buy @WasabiFan a :hamburger:! Make that 10 :hamburger:s and some :fries:, lots of :fries:. (In other words thank you, thank you, thank you - I know this was a lot of work!)

ddemidov commented:
This looks awesome!

WasabiFan commented:
> In other words thank you, thank you, thank you - I know this was a lot of work!

This type of work is pretty enjoyable, so I don't really consider it work :wink: Historically, I have written code for websites, but this is the first time that I made such large styling additions (essentially starting with just the infrastructure and content) without a template to work off of -- It's nice to get that kind of experience. And I'm pretty pleased with how it turned out!

rhempel commented:
Put it on your resume :-)

JorgePe commented:
Great, great work! Congrats!

ddemidov commented:
It looks like http://www.ev3dev.org/docs/libraries/ is no longer accessible from anywhere on the new site.

ddemidov commented:
No, I am wrong; it's under Docs/More.
